File: src/main/java/com/randomperson22/piggymodtest/entities/umbra/UmbraModel.java
package com.randomperson22.piggymodtest.entities.umbra;

import net.minecraft.util.ResourceLocation;
import software.bernie.geckolib3.model.AnimatedGeoModel;

public class UmbraModel extends AnimatedGeoModel<EntityUmbra> {
	
    public ResourceLocation getModelLocation(EntityUmbra entity) {
        return new ResourceLocation("piggymodtest", "geo/umbra.geo.json");
    }

    public ResourceLocation getTextureLocation(EntityUmbra entity) {
        return new ResourceLocation("piggymodtest", "textures/entities/umbra.png");
    }

    public ResourceLocation getAnimationFileLocation(EntityUmbra entity) {
        return new ResourceLocation("piggymodtest", "animations/umbra.animation.json");
    }
}

File: src/main/kotlin/com/randomperson22/piggymodtest/entities/umbra/UmbraModel.kt
package com.randomperson22.piggymodtest.entities.umbra

import net.minecraft.util.ResourceLocation
import software.bernie.geckolib3.model.AnimatedGeoModel

class UmbraModel : AnimatedGeoModel<EntityUmbra?>() {
    override fun getModelLocation(entity: EntityUmbra?): ResourceLocation {
        return ResourceLocation("piggymodtest", "geo/umbra.geo.json")
    }

    override fun getTextureLocation(entity: EntityUmbra?): ResourceLocation {
        return ResourceLocation("piggymodtest", "textures/entities/umbra.png")
    }

    override fun getAnimationFileLocation(entity: EntityUmbra?): ResourceLocation {
        return ResourceLocation("piggymodtest", "animations/umbra.animation.json")
    }
}
